mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,394 @@
+
+**Note:** If you were to write the email tag self-closing (``<email mail-to="Rick" />``), the final output would also be self-closing. To enable the ability to write the tag with only a start tag ( <email mail-to="Rick"> ) you must decorate the class with the following:
+
+.. literalinclude:: authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/TagHelpers/EmailTagHelperMailVoid.cs
+   :lines: 6
+   :dedent: 4 
+   
+In our example, the output would be ``<a href="mailto:Rick@contoso.com" />``. Self-closing anchor tags are not valid HTML, so you wouldn't want to create one, but you might want to create a tag helper that is self-closing. The ASP.NET 5 runtime sets the type of the ``TagMode`` property after reading a tag.
+
+.. In this section we will update the ``EmailTagHelper`` so that it gets the target ``mail-to`` from the content. Need to revert the contact.cshtml file back.
+
+An asynchronous email helper
+_____________________________________
+
+In this section we'll write an asynchronous email helper. 
+
+#. Replace the ``EmailTagHelper`` class with the following code:
+
+.. literalinclude:: authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/TagHelpers/EmailTagHelper.cs
+   :lines: 6-17
+   :dedent: 4
+
+**Notes:**
+
+- This version uses the asynchronous ``ProcessAsync`` method. The asynchronous ``GetChildContentAsync`` returns a ``Task`` containing the ``TagHelperContent``. 
+- We use the ``context`` parameter to get contents of the HTML element.
+
+2. Make the following change to the *Views/Home/Contact.cshtml* file so the tag helper can get the target email.
+
+.. literalinclude::  authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/Views/Home/Contact.cshtml
+   :language: aspx-cs
+   :emphasize-lines: 15-16 
+   :lines: 1-17
+
+3. Run the app and verify that it generates a valid email link.
+    
+The bold Tag Helper
+---------------------------
+
+#. Add the following ``BoldTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/TagHelpers/BoldTagHelper.cs
+    :language: c#
+
+**Notes:** 
+
+- The ``[TargetElement]`` attribute passes an attribute parameter that specifies that any HTML element that contains an HTML attribute named "bold" will match, and the ``Process`` override method in the class will run. In our sample, the ``Process``  method removes the "bold" attribute and surrounds the containing markup with ``<strong></strong>``.
+-  Because we don't want to replace the existing tag content, we must write the opening ``<strong>`` tag with the ``PreContent.SetContent`` method and the closing ``</strong>`` tag with the ``PostContent.SetContent`` method.
+
+2. Modify the *About.cshtml* view to contain a ``bold`` attribute value. The completed code is shown below.
+
+.. literalinclude::  authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/Views/Home/AboutBoldOnly.cshtml
```

Comment:
This is just one spot in one file, right? Should be pretty quick to find all the literal includes for just this one file and offset the line numbers. It looks weird, so I think we should fix it.